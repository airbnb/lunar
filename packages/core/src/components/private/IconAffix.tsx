// Required for TS
// eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
import React from 'react';
import withStyles from '../../composers/withStyles';
import BaseAffix, { Props as BaseProps } from './BaseAffix';

export type Props = BaseProps;

export default withStyles(({ unit, pattern }) => ({
  affix: {
    display: 'inline-block',
    verticalAlign: 'middle',
    // The icon doesn't align perfectly in the middle unless the parent is flex,
    // but most (if not all) parents are inline-block, so we need to bump it up a few pixels.
    marginTop: -(unit / 3.25),
  },

  affix_after: {
    marginLeft: unit,
  },

  affix_before: {
    marginRight: unit,
  },

  affix_disabled: {
    ...pattern.disabled,
  },

  affix_flex: {
    flexShrink: 0,
    flexGrow: 0,
    marginTop: 0,
  },
}))(BaseAffix);
