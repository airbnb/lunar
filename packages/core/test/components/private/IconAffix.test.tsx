import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import IconAffix from '../../../src/components/private/IconAffix';

describe('<IconAffix />', () => {
  it('renders before', () => {
    const wrapper = shallowWithStyles(<IconAffix before>Child</IconAffix>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders after', () => {
    const wrapper = shallowWithStyles(<IconAffix after>Child</IconAffix>);

    expect(wrapper).toMatchSnapshot();
  });

  it('errors if both props used', () => {
    expect(() => {
      shallowWithStyles(
        <IconAffix before after>
          Child
        </IconAffix>,
      );
    }).toThrowError();
  });
});
