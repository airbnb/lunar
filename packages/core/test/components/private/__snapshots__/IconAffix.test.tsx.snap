// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconAffix /> renders after 1`] = `
<div
  className="affix_zg9aho-o_O-affix_after_pog3hg"
>
  Child
</div>
`;

exports[`<IconAffix /> renders before 1`] = `
<div
  className="affix_zg9aho-o_O-affix_before_14tkmhr"
>
  Child
</div>
`;
